internal/internal_task_pollers.go

```diff
@@ -279,15 +285,17 @@ func newWorkflowTaskPoller(
 	params workerExecutionParameters,
 ) *workflowTaskPoller {
 	return &workflowTaskPoller{
-		basePoller:                   basePoller{shutdownC: params.WorkerStopChannel},
+		basePoller: basePoller{
+			shutdownC:    params.WorkerStopChannel,
+			metricsScope: metrics.NewTaggedScope(params.MetricsScope),
```

Comment:
I think this should be tagged by domain. So we can create an alert on our side capturing both the service failing and the domain that is affected.

Reply:
The metrics emitted by this `metricsScope` already has domain, tasklist and service tags.